message.go

```diff
@@ -229,8 +230,24 @@ type MessageEmbed struct {
 	Provider    *MessageEmbedProvider  `json:"provider,omitempty"`
 	Author      *MessageEmbedAuthor    `json:"author,omitempty"`
 	Fields      []*MessageEmbedField   `json:"fields,omitempty"`
+	
+	//flag that tells the marshaller to marshal struct as nil
+	marshalnil  bool		   `json:"-"`
+}
+
+func (e *MessageEmbed) MarshalNil(flag bool) *MessageEmbed {
+	e.marshalnil = flag
+	return e
+
 }
 
+func (e *MessageEmbed) MarshalJSON() ([]byte, error) {
+	if e.marshalnil == true {
+		return json.Marshal(nil)
+	}
+	type EmbedAlias MessageEmbed
+	return json.Marshal(&struct{*EmbedAlias}{EmbedAlias: (*EmbedAlias)(e),})
```

Comment:
Why are you doing this type alias and anonymous struct thing? why not just `json.Marshal(e)`?

Reply:
As discussed in dm. direct is not possible. If you want to avoid the anonymous struct since we are not changing any field here, this can also be used :
```
outer_e := EmbedAlias(*e)
return json.Marshal(&outer_e)
```
Let me know what you prefer. The struct method is somewhat more generic i guess because it lets you modify any fields if necessary.